force-app/main/default/classes/VariablesDataTypesOperators.cls

```diff
@@ -259,25 +378,35 @@ public with sharing class VariablesDataTypesOperators {
      */
     public static Integer performDivisionAndCast(Double a, Double b) {
         // Perform the division and cast (round down) off the result.
-        Double divisionResult = null;
+        If( a== null || b == null)
+        {
+            return null;
+        }
+        
+        Double divisionResult = (a/b);
 
         // Write the code for type casting the divisionResult to an Integer
-        Integer roundedResult = null;
-        return null; // Replace null with the variable you used to store the result
+        Integer roundedResult = Integer.valueOf(divisionResult);
+        return roundedResult; // Replace null with the variable you used to store the result
     }
 
     /**
      * Question 18
-     * Calculate the weekly and monthly paycheck based on hourly rate and number of hours.
+     * Calculate the weekly paycheck based on hourly rate and number of hours.
      * Weekly Paycheck is calculated by the formula: rate multiplied by hours.
      * Example: calculateWeeklyPaycheck(10.0, 40.0) should return 400.0
      * @param hourlyRate The rate of pay per hour.
      * @param numberOfHours The number of hours worked in a week.
      * @return A Decimal representing the weekly paycheck.
      */
     public static Decimal calculateWeeklyPaycheck(Decimal hourlyRate, Double numberOfHours) {
+        If (hourlyRate == null || numberOfHours == null)
+            { 
+            return null;
+               }
+          Decimal weeklyPayCheck = (hourlyRate * Decimal.valueOf(numberOfHours)).setScale(2, RoundingMode.HALF_UP);
```

Comment:
This looks great and it seemed like you went through a lot of research to get this answer.
Would you say that this is a little easier to understand?

 Decimal weeklyPaycheck = hourlyRate * numberOfHours;
try it and see if it works the same.